test/extended/machines/workers.go

```diff
@@ -122,30 +121,34 @@ var _ = g.Describe("[Feature:Machines][Disruptive] Managed cluster should", func
 		skipUnlessMachineAPIOperator(c.CoreV1().Namespaces())
 
 		g.By("validating node and machine invariants")
-		// fetch machines
-		machines, err := listMachines(dc, machineLabelSelectorWorker)
+		// fetch all machines
+		allMachines, err := listMachines(dc, "")
 		if err != nil {
 			e2e.Failf("unable to fetch worker machines: %v", err)
 		}
-		numMachineWorkers := len(machines)
-		if numMachineWorkers == 0 {
-			e2e.Failf("cluster should have worker machines")
+
+		if len(allMachines) == 0 {
+			e2e.Failf("cluster should have machines")
 		}
 
 		// fetch nodes
 		nodes, err := c.CoreV1().Nodes().List(metav1.ListOptions{
 			LabelSelector: nodeLabelSelectorWorker,
 		})
 		o.Expect(err).NotTo(o.HaveOccurred())
+		if len(nodes.Items) == 0 {
+			e2e.Failf("cluster should have nodes")
+		}
 		// map node -> machine
-		nodeToMachine, nodeMatch := mapNodeNameToMachineName(nodes.Items, machines)
+		nodeToMachine, nodeMatch := mapNodeNameToMachineName(nodes.Items, allMachines)
 		if !nodeMatch {
 			e2e.Failf("unable to map every node to machine.  nodeToMachine: %v, nodeName: %v", nodeToMachine, nodeNames(nodes.Items))
 		}
-		machineToNode, machineMatch := mapMachineNameToNodeName(machines, nodes.Items)
-		if !machineMatch {
-			e2e.Failf("unable to map every machine to node.  machineToNode: %v, machineNames: %v", machineToNode, machineNames(machines))
+		machines := make([]objx.Map, len(nodeToMachine))
+		for _, m := range nodeToMachine {
+			machines = append(machines, m)
 		}
+		numMachineWorkers := len(machines)
```

Comment:
This will also include master machines. So `numMachineWorkers` variable name is not correct.

Additionally, for machine loop on line 154 will then delete all machines, including masters. Unless, there is something obvious I don't see.

Reply:
No, these machines will not include master machines. These machines are collected using worker nodes.

Reply:
I see now. Only worker nodes are pulled.